### ISSUE TITLE
tlsOptions are ignored

### ISSUE THREAD

**Issue by alvestjo** [jsumners/node-activedirectory#5]
Hi,

It seems the tlsOptions parameter is ignored and not passed to ldapjs. I am using v1.1.0. 
//jonas

**Comment by jsumners** [comment]
Are you passing the options at construction time or via a method?

**Comment by alvestjo** [comment]
I am passing it to the constructor.
//jonas

**Comment by jsumners** [comment]
Can you provide a sample of code?

**Comment by alvestjo** [comment]
Hi,

I am using it like this (like it is documented in original ActiveDirectory fork - I now noticed that this option is not documented in your fork):
---------------------------------
var ActiveDirectory = require('activedirectory2');
var config = { url: 'ldaps://dc.domain.com:636',
               baseDN: 'dc=domain,dc=com',
               username: 'username@domain.com',
               password: 'password',
               tlsOptions: {
                   rejectUnauthorized: false
               },
 }
var ad = new ActiveDirectory(config);
----------------------------------------

I think the problem is tlsOptions is not passed to the  Object.defineProperties() call on line 207 in ActiveDirectory.js. 
//jonas

**Comment by jsumners** [comment]
Thanks. I'll work on this as soon as I can.

**Comment by jsumners** [comment]
Please try commit aff2faa and let me know if it works for you:

```
$ npm install jsumners/node-activedirectory#aff2faa
```

**Comment by jurjendijkstra-vma** [comment]
Hi James,
It almost works :-) 
in components/utilities.js, line 24 you have:
```javascript
  const ldapOpts = getLdapClientOpts(Object.assign({ url: _url }, opts))
```
the tlsOptions that we want to pass however are not in opts, they are in ad.opts
so I would propose:
```javascript
  const ldapOpts = getLdapClientOpts(Object.assign({ url: _url }, opts, ad.opts))
```
or perhaps the parameters in reversed order, I am not sure:
```javascript
  const ldapOpts = getLdapClientOpts(Object.assign({ url: _url }, ad.opts, opts))
```

then it works alright for me.

**Comment by jsumners** [comment]
@jurjendijkstra-vma the correct one would be

```
const ldapOpts = getLdapClientOpts(Object.assign({ url: _url }, ad.opts, opts))
```

Would you like to submit a PR?

**Comment by jurjendijkstra-vma** [comment]
Sure, I am glad to submit a PR.  This is it:  https://github.com/jsumners/node-activedirectory/pull/6

**Comment by jsumners** [comment]
Closed in 1.2.1.

**Comment by jurjendijkstra-vma** [comment]
Thanks for the pull and release :-)
